pkg/cluster/compute/affinityrule/affinityrule.go

```diff
@@ -97,3 +98,45 @@ func GetAffinityRule(whost *host.WmiHost, affinityRuleName string) (caffinityRul
 	caffinityRule = &AffinityRule{wmiafRule}
 	return
 }
+
+// SetAffinityRule updates an existing affinity rule
```

Comment:
you couldn't merge it in existing 'CreateAffinityRule' method (rather than creating another function)?

Reply:
Just to be consistent with current code, added a simple wrapper on WMI method and have specific logic in moc-pkg layer. I will update as suggested

Reply:
Refactored as suggested in new patch